README
Copyright 2007, 2008, 2009 Kevin Ryde

This file is part of Gtk2-Ex-Clock.

Gtk2-Ex-Clock is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

Gtk2-Ex-Clock is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License along
with Gtk2-Ex-Clock.  If not, see <http://www.gnu.org/licenses/>.




Gtk2::Ex::Clock is a simple digital clock widget displaying an
strftime formatted time or date/time.  The default is 24-hour %H:%M
local time, and properties allow other formats and/or a different
timezone.  Pango markup can be used for font effects like bold or
italics in the display.

The only prerequisite is Gtk2-Perl.  The clock home page is

    http://user42.tuxfamily.org/gtk2-ex-clock/index.html

Bug reports to the author at user42@zip.com.au please.


Changes
Copyright 2007, 2008, 2009, 2010 Kevin Ryde

This file is part of Gtk2-Ex-Clock.

Gtk2-Ex-Clock is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

Gtk2-Ex-Clock is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License along
with Gtk2-Ex-Clock.  If not, see <http://www.gnu.org/licenses/>.



Version 14, October 2010
    - tests check for DateTime as well as DateTime::TimeZone

Version 13, September 2010
    - new "timezone-string" property

Version 12, May 2010
    - for minute resolution allow minutes not on a time() % 60
      boundary, as happens for DateTime::TimeZone::TAI and similar

Version 11, April 2010
    - use POSIX::Wide, fixes netbsd locale codeset "646"

Version 10, February 2010
    - new home page

Version 9, June 2009
    - all wide chars in format string, irrespective of locale

Version 8, June 2009
    - more example programs

Version 7, November 2008
    - use Tie::TZ, use Gtk2 1.200

Version 6, August 2008
    - need explicit tzset for threaded perl 5.8.8 

Version 5, July 2008
    - recognise DateTime %{} formats with seconds resolution
    - let perl take care of tzset if it's needed on old systems

Version 4, June 2008
    - fix for _SC_CLK_TCK on perl 5.8.8

Version 3, May 2008
    - cope with locale charset from POSIX::strftime()

Version 2, December 2007
    - tweaks

Version 1, November 2007
    - the first version


Makefile.PL
#!/usr/bin/perl

# Copyright 2007, 2008, 2009, 2010 Kevin Ryde

# This file is part of Gtk2-Ex-Clock.
#
# Gtk2-Ex-Clock is free software; you can redistribute it and/or modify it
# under the terms of the GNU General Public License as published by the Free
# Software Foundation; either version 3, or (at your option) any later
# version.
#
# Gtk2-Ex-Clock is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
# for more details.
#
# You should have received a copy of the GNU General Public License along
# with Gtk2-Ex-Clock.  If not, see <http://www.gnu.org/licenses/>.

use 5.008;
use strict;
use warnings;
use ExtUtils::MakeMaker;
use lib 'inc';
use MyMakeMakerExtras;

MyMakeMakerExtras::WriteMakefile
  (NAME         => 'Gtk2-Ex-Clock',
   ABSTRACT     => 'Simple digital clock widget.',
   VERSION_FROM => 'lib/Gtk2/Ex/Clock.pm',
   AUTHOR       => 'Kevin Ryde <user42@zip.com.au>',
   LICENSE      => 'gpl',
   SIGN         => 1,
   MIN_PERL_VERSION => '5.008',

   PREREQ_PM    => {
                    # Gtk2 version 1.200 for GDK_PRIORITY_REDRAW
                    # (demanding a version just one constant is pretty
                    # gratuitous, but you'll almost certainly it for various
                    # bug fixes too)
                    'Gtk2' => '1.200',
                    'Glib::Object::Subclass' => 0,

                    'Glib::Ex::SourceIds' => 2,  # ObjectBits v.2

                    # actually this is only for the string timezone case,
                    # but it's small enough to grab always
                    'Tie::TZ' => 0,

                    # version 2 for netbsd 646 alias
                    'POSIX::Wide' => 2,
                   },

   META_MERGE
   => { resources
        => { homepage
             => 'http://user42.tuxfamily.org/gtk2-ex-clock/index.html',
           },
        recommends
        => { 'DateTime'           => 0,
             'DateTime::TimeZone' => 0,
           },
        optional_features
        => { with_datetime
             => { description => 'Have "make test" do as much as possible.',
                  requires => { 'DateTime'           => 0,
                                'DateTime::TimeZone' => 0,
                              },
                },
             maximum_tests
             => { description => 'Have "make test" do as much as possible.',
                  requires => { 'Test::Weaken' => '2.000',
                              },
                },
             maximum_examples
             => { description => 'Be able to run all the example programs.',
                  requires => { 'DateTime'                => 0,
                                'DateTime::TimeZone'      => 0,
                                'DateTime::TimeZone::TAI' => 0,
                                'DateTimeX::AATW'         => 0,
                              },
                },
             maximum_devel
             => { description => 'Stuff used variously for development.',
                  requires => { 'Glib::Ex::ConnectProperties' => 0,
                                'Glib::Ex::SignalIds'         => 0,
                                'Devel::FindRef'              => 0,
                                'DateTime'                    => 0,
                                'DateTime::TimeZone'          => 0,
                                'DateTime::TimeZone::TAI'     => 0,
                                'Time::Mock'                  => 0,
                              },

                },
           },
      },

   MyMakeMakerExtras_Pod_Coverage
   => [ 'Gtk2::Ex::Clock,also_private,[qr/^[A-Z_]+$$/]' ],
  );
